packages/web/src/ducks/events/sagas/event.sagas.ts

```diff
@@ -164,10 +164,15 @@ export function* deleteSomedayEvent({ payload }: Action_DeleteEvent) {
 }
 
 export function* editEvent({ payload }: Action_EditEvent) {
+  const { _id, applyTo, event, shouldRemove } = payload;
+
   try {
-    yield put(eventsEntitiesSlice.actions.edit(payload));
-    yield call(EventApi.edit, payload._id, payload.event, {
-      applyTo: payload.applyTo,
+    shouldRemove
+      ? yield put(eventsEntitiesSlice.actions.delete({ _id }))
+      : yield put(eventsEntitiesSlice.actions.edit(payload));
+
+    yield call(EventApi.edit, _id, event, {
+      applyTo: applyTo,
     });
```

Comment:
I'm using the `shouldRemove` property to determine whether to call `.edit` or `.delete` (previously they were both called sequentially). 


Technically, we should also be removing the id from the `getWeekEventsSlice` if `shouldRemove` is true. That is so the week slice accurately depicts what is in view. However, I found that to cause the event to stay on the screen. Instead of going down that rabbit hole, I'm OK introducing this as a code smell. In the long term, I'd like to replace Redux Saga (and possibly Redux) with something less verbose and complex. So, I don't want to invest a lot more into getting everything with Redux and Redux Saga to work perfectly.

Reply:
Imo keeping redux is fine for global state, maybe we could just simplify the store a bit, and let the requests and async work outside the store, or use Redux Thunks for that.